docs/design/2021-04-21-unify-log-library.md

```diff
@@ -0,0 +1,141 @@
+
+Then, determine whether the configuration has enabled slow query log, and if so, create a log handler specific to slow query.
+
+[Here is the code](https://github.com/pingcap/tidb/blob/e79fa8c6b654e5b94e9ed0a1c0f997d6564e95be/util/logutil/log.go#L261).
+
+Regarding where these two handlers are used.
+
+- Some historical legacy code, such as `cmd/importer/parser.go`, which uses the standard logger by `logrus`.
+- Slow query log all uses the slow query log handler created by `logrus`, code in `executor/adapter.go`.
+
+#### [pingcap/log](https://github.com/pingcap/log)
+
+`pingcap/log` is a wrapper around zap, and as mentioned below the two terms are equivalently interchangeable.
+
+Similar to `logrus`, the init method of zap `func InitZapLogger(cfg *LogConfig) error` may initialize two zap handlers.
+
+- The global zap handler, the default log handler for the entire repo, through which the vast majority of logs are emitted.
+- Slow query zap handler, which is only initialized and not used.
+  
+`InitZapLogger`'s logic is very similar to `logrus`' above.
+
+#### gRPC Logger
+
+I almost forgot that there is a fish in the net, which is not in `util/logutil/log.go`. In `main.go` there is a bunch of grpc logger initialization code.
+
+[Here is the code](https://github.com/pingcap/tidb/blob/e79fa8c6b654e5b94e9ed0a1c0f997d6564e95be/tidb-server/main.go#L591).
+
+The `NewLoggerV2` method creates a go native logger handler and is only used in grpc.
+
+### PD
+
+PD is similar to TiDB in that it also relies on `logrus` and `pingcap/log`, but with an additional layer of `capnslog` as a proxy. 
+
+Log library dependencies:
+
+![pd-log-dependency](./imgs/pd-log-dependency.png)
+
+#### Logrus
+
+The standard logger is then passed down to the etcd (via the `capnslog` proxy), grpc and draft components as the log handler for these packages.
+
+There is only one `logrus` handler inside the entire PD codebase.
+
+Only the etcd, grpc, and draft components use the `logrus` handler.
+
+The initialization of `logrus` locates at `pkg/logutil/log.go`. [Here is the code](https://github.com/tikv/pd/blob/b07be86fb91aef07e8a68258ff6149256ab511f8/pkg/logutil/log.go#L260).
+
+#### [pingcap/log](https://github.com/pingcap/log)
+
+There is only one zap log handler inside the entire PD codebase, and its initialization is inline `cmd/pd-server/main.go`.
+
+[Here is the code](https://github.com/tikv/pd/blob/b07be86fb91aef07e8a68258ff6149256ab511f8/cmd/pd-server/main.go#L66).
+
+The logic is simple, create a new handler based on the configuration and replace the global handler at the `pingcap/log` package level.
+
+Most of the logging logic in PD will use the global zap handler.
+
+### [pingcap/br](https://github.com/pingcap/br)
+
+BR is strangely listed here, it is not a necessary component for the tidb cluster to run. It is here just for one reason: tidb depends on BR, which in turn depends on tidb's `util/logutil/log.go`, constituting a circular dependency. 
```

Comment:
BR is nested in TiDB now. So `it is not a necessary component for the tidb cluster to run.` is not accurate.